source/includes/library-reference/_library-reference.html.md

```diff
@@ -0,0 +1,110 @@
+inherit from `bonsai.Simulator` that are not specified below are eligible
+objectives (i.e. reward functions).
+
+
+### start()
+
+Called by the AI Engine to start the simulator at the start of a train or test
+pass.  `start` is called for the first time at the beginning of training and is
+also the last of the stop, set_properties, reset, start message sequence sent
+from the AI Engine in between training and test passes.
+
+
+### stop()
+
+Called by the AI Engine to stop training or testing. Training is also the first
+callback of the stop, set_properties, reset, start message sequence sent by the
+AI Engine between training and test passes.
+
+
+### reset()
+
+Called by the AI Engine to reset simulator state in between training and test
+passes.
+
+
+### set_properties(**kwargs)
+
+Called by the AI Engine. to configure the simulation before training is started
+or restarted. `set_properties` is called for the first time at the beginning of
+training and is also called as part of the stop, set_properties, reset, start
+message sequence sent from the AI Engine between training and test passes. See
+the configure clause of the lesson statement in this simulator's accompanying
+curriculums to get more information about the values in `kwargs`.
+
+
+### advance(actions)
+
+Advance the simulation forward one tick. `actions` contains a dictionary of key
+values as defined by this simulator's action schema in Inkling.
+
+
+### get_state()
+
+Returns a SimState containing the state of the simulation computed after the
+last call to `advance`. state is a dictionary matching the state schema as
+defined in an Inkling `simulator` statement within an inkling program.
+is_terminal is only true when the simulator is in a "game over" state. Any
+cleanup of the is_terminal state should be handled by advance or get_state.
+
+
+### objectives (reward functions)
+
+Any methods added to this class that are not named with one of the above
+method names are eligible to be called as a reward function.  To be called,
+the name of the function specified in an Inkling `lesson` must match the name
+of the method.  Reward functions specified in Inkling but not implemented on a
+`Simulator` class will result in an error.
+
+## SimState
+
+A named tuple returned by `Simulator.get_state`
+
+
+### state
+
+A dictionary matching the state schema as defined in an Inkling `simulator`
+statement within an inkling program
+
+
+### is_terminal
+
+`True` when the simulator is in a "game over" state otherwise `False`.
+
+
+## run
+
+### bonsai.run_for_training_or_prediction(name, simulator_or_generator)
+
+Connects to the AI Engine and idenitifies as `name`. The `name` of this
+simulator must be the same in Inkling as it is in this function call.
+`simulator_or_generator` is a reference to an instance of the simulator. This
+call blocks and drives the simulator loop until training stops.
+
+
+## Example
+
+For more information, see [examples][1].

```

Comment:
Only the Find the Center example has a simulator example, should we link to that one specifically?